### PR TITLE
Convert more classes in test_crypto.py to pytest-style

### PR DESCRIPTION
**Pull request by alexwlchan**
Continuing to break up #561.

### CODE REVIEW

File: tests/test_crypto.py [hunk 1]
```diff
@@ -878,116 +870,108 @@ def test_failedGeneration(self):
...
+        with pytest.raises(TypeError):
+            pub.check()
 
 
-class X509NameTests(TestCase):
-    """
-    Unit tests for :py:class:`OpenSSL.crypto.X509Name`.
-    """
+def x509_name(**attrs):
+    """Return a new X509Name with the given attributes."""
```
Comment (hynek) [inline tests/test_crypto.py]: Newlines after/before `"""` please.

File: tests/test_crypto.py [hunk 2]
```diff
@@ -806,59 +806,50 @@ def test_convert_private_pkey_to_cryptography_key(self):
...
         key = PKey()
-        self.assertEqual(key.type(), 0)
-        self.assertEqual(key.bits(), 0)
-        self.assertRaises(TypeError, key.check)
+        assert key.type() == 0
+        assert key.bits() == 0
+        with pytest.raises(TypeError):
+            key.check()
 
     def test_failedGeneration(self):
```
Comment (hynek) [inline tests/test_crypto.py]: Could you pep8-ize all the camelCases?

**Comment by alexwlchan** [comment]
Markups made.

**Comment by alexwlchan** [comment]
🔥🔥 Intermittent CI failures 🔥🔥

**Comment by hynek** [comment]
🥇

**Pull request merged**